ci-operator/step-registry/cucushift/installer/check/gcp/machine-type/cucushift-installer-check-gcp-machine-type-commands.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+expected_control_plane_type="$(yq-go r ${CONFIG} controlPlane.platform.gcp.type)"
+expected_compute_type="$(yq-go r ${CONFIG} compute[0].platform.gcp.type)"
```

Comment:
I suggest to get machine types from `CONTROL_PLANE_NODE_TYPE` and `COMPUTE_NODE_TYPE`, `CONFIG` is a intermediate file after `CONTROL_PLANE_NODE_TYPE` and `COMPUTE_NODE_TYPE` are received, as a check point step, I prefer to use the ENV vars directly.

Reply:
In fact, it was my first thinking too. But, on considering the step shall be available for any GCP workflow, which possibly doesn't specify `CONTROL_PLANE_NODE_TYPE` and etc., I changed to fetch from install-config.yaml instead.

Reply:
According to the [lines](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L474-L489) in ipi-install-install, the file `"${SHARED_DIR}/install-config.yaml"` does be used for the IPI installation, and also untouched during the installation. I wonder what's your concern exactly, thanks!

Reply:
> the lines in ipi-install-install

1. Actually when I was reviewing the codes in the step for the 1st time, I do not understand why it saved a copy of install-config.yaml, from my personal review, `"${SHARED_DIR}/install-config.yaml"` should be never saved for the beginning. 
2. Because the install-config.yaml might be override again and again by other steps, once someone introduced some script bug, the install-config.yaml would become buggy, so I prefer to not relay on the file, but ENV vars.
3. For your concern - `which possibly doesn't specify CONTROL_PLANE_NODE_TYPE and etc`, I think we can judge the ENV vars empty or not, then move forwards the checking.